ProjFS.Mac/PrjFSKextLogDaemon/PrjFSKextLogDaemon.cpp

```diff
@@ -221,12 +239,141 @@ static bool TryFetchAndLogKextHealthData(io_connect_t connection)
             healthData.totalFindRootForVnodeMisses,
             healthData.totalRefreshRootForVnode,
             healthData.totalInvalidateVnodeRoot);
+        
+        WriteJsonToMessageListener(
```

Comment:
It’s hard to see the usage (here on my iPad) but I wonder if you want to pass the event name rather than have the message field be the determinant.  That is, pass eventName as “kext.health” for example. This would arrive to kusto as “vfs.kext.health”.  Then kusto queries would see it as a top-level field.

But I’m not sure how many message types you have. 

I thought I saw an error message too.  Maybe have it be eventName=“kext.error” and let it have a “message” and “ret” fields.